In [3]:
import random
from tqdm import tqdm

The theoretical probability of eating a hot dorito for the first round of roulette is 1/6. 

In [4]:
def roulette(simcount=9000000):
    with tqdm(total=simcount, desc="Processing...") as pbar:
        results=0
        for i in range(simcount):
            if random.randint(1,6) == 6:
                results+=1
            pbar.update(1)
        final=(results/simcount)*100
    return final, simcount

pain, suffering=roulette()
print(f'After {suffering} amount of trials, we have found that on average you have a {pain}% chance of having the painful dorrito.')

Processing...: 100%|██████████| 9000000/9000000 [00:12<00:00, 707411.39it/s]

After 9000000 amount of trials, we have found that on average you have a 16.680644444444447% chance of having the painful dorrito.


After running a significant amount of trials, the empirical probability should approach the theoretical probability due to the law of large numbers. My empirical probability is 16.6806% which is very close to the 16.6666% theoretical probability.